src/compiler/checker.ts

```diff
@@ -15473,13 +15603,27 @@ namespace ts {
                 bindSourceFile(file, compilerOptions);
             });
 
+            let mergeAugmentations = false;
             // Initialize global symbol table
             forEach(host.getSourceFiles(), file => {

```

Comment:
it can and I think we should make a sweep over the compiler to convert `forEach` to `for..of` at some point. But this should be a separate PR.